### ISSUE TITLE
Optimizing Serializers for ACSets

### ISSUE THREAD

**Issue by jpfairbanks** [AlgebraicJulia/Catlab.jl#484]
We currently have JSON serializers for ACSets #265, but when performance gets critical we are going to want to optimize these and the construction of the indexing structures. The way this works for Graphs is 

1. read an edge list off disk
2. sort and count to get all the neighborhood sizes,
3. allocate all the indexes exactly
4. fill all the data with bulk copies

if constructor performance because a critical application bottleneck, we could design a serializer/deserializer that took a schema and generated the optimal binary format for it, along with a serializer/deserializer for that format.

**Comment by epatters** [comment]
We might also consider integrating with something like Apache Arrow rather than writing our own binary format.

**Comment by jpfairbanks** [comment]
Yeah, the [Arrow.jl](https://github.com/JuliaData/Arrow.jl) package seems to be getting reasonable maintenance, and have the kinds of types we would want for ACSets to interop with RDBMSes.
